### ISSUE TITLE
Mixed precision multi shift issues

### ISSUE THREAD

**Issue by mathiaswagner** [lattice/quda#204]
While looking at the mixed precision implementation of the multi-shift I noticed a couple of possible weak points:

If the reliable update detects an increasing residual the multi shift is stopped using ```break```. It probably would be better to just stop working on that shift.
Check is done right now like
```C++
	  if (resIncrease > maxResIncrease or resIncreaseTotal[reliable_shift] > maxResIncreaseTotal) break; // check if we reached the limit of our tolerance break;
```

At the end of the reliable update the reliable update parameters are only update for the shift that triggered the update:
```C++
	int m = reliable_shift;
	rNorm[m] = sqrt(r2[0]) * zeta[m];
	maxrr[m] = rNorm[m];
	maxrx[m] = rNorm[m];
	r0Norm[m] = rNorm[m];      
```

As we update the residual for all shifts (that are still worked on) it might be better to just do that for all shifts up to ```num_offset_now```. 

It also seems like other than for the CG not reliable update is triggered if we are seeing convergence.

Still looking into some more details.

**Comment by mathiaswagner** [comment]
Updating the reliable update parameters for all shifts instead of only for the `reliable_shift` seems to reduce the number of reliable update used. For double/ single I saw a reduction from 76 to 24 for the testcase described in #205. True residual were not affected in this case.

**Comment by mikeaclark** [comment]
Yes that sounds familiar.  Reducing unnecessary reliable updates will likely improve overall performance so is worth doing.  Are you seeing an actual speedup?

**Comment by mathiaswagner** [comment]
Number of iterations is constant and right now I am killing the job right after the multishift and don’t do timings. But I guess overall it does not matter so much (18000 iterations and 24 vs 76 rel updates). Will do some timings later.

**Comment by mathiaswagner** [comment]
I have implemented both suggestions in the mw/mixedmultishift branch (Commits fb6642ab14f0c219617ee83dfb5d171471b42199 and 9abbab127dd90953076c8787f80f40dddd90640c).
That closes this issue but leaves #205 open.
